src/components/molecules/ChatMessageBox/ChatMessageBox.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPaperPlane } from "@fortawesome/free-solid-svg-icons";
+
+import "./ChatMessageBox.scss";
+
+export interface ChatMessageBoxProps {
+  sendMessage: (text: string) => void;
+}
+
+export const ChatMessageBox: React.FC<ChatMessageBoxProps> = ({
+  sendMessage,
+}) => {
+  const [isSendingMessage, setMessageSending] = useState(false);
+
+  // This logic dissallows users to spam into the chat. There should be a delay, between each message
+  useEffect(() => {
+    if (isSendingMessage) {
+      setTimeout(() => {
+        setMessageSending(false);
+      }, 500);
+    }
+  }, [isSendingMessage]);
```

Comment:
Cleaning up the timeout within the useEffect's cleanup function + using a `const` for the timeout rather than a 'magic number' (which should be defined in `src/settings.ts`

```suggestion
  useEffect(() => {
    if (!isSendingMessage) return;
    
    const timeoutId = setTimeout(() => {
      setMessageSending(false);
    }, CHAT_MESSAGE_TIMEOUT);
    
    return () => {
      clearTimeout(timeoutId)
    }
  }, [isSendingMessage]);
```

- https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup
  - https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1
- https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout
- https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/clearTimeout

Reply:
Potentially could also use `react-use`'s `useTimeoutFn` here maybe?

- https://github.com/streamich/react-use/blob/master/docs/useTimeoutFn.md